File: Week 6 - Python/cash/cash.py
from cs50 import get_float

def main():
    while True:
        answer = get_float("Change: ")
        if answer > 0:
            break

    calculate_coins(answer)


def calculate_coins(cash):
    quarters = dimes = nickels = pennies = 0
    cash = cash * 100
    cash = int(cash)

    # Check for quarters
    if cash >= 25:
        quarters = int(cash / 25)
        cash = cash - quarters * 25

    # Check for dimes
    if 25 > cash >= 10:
        dimes = int(cash / 10)
        cash = cash - dimes * 10

    # Check for nickels
    if 10 > cash >= 5:
        nickels = int(cash / 5)
        cash = cash - nickels * 5

    # Check for pennies
    if 5 > cash >= 1:
        pennies = cash

    amount = quarters + dimes + nickels + pennies
    print(amount)


if __name__ == "__main__":
    main()

File: Week 1 - C/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(int argc, char *argv[])
{
    int change = 0, quarters = 0, dimes = 0, nickels = 0, pennies = 0;

    do
    {
        // Prompt for change
        change = get_int("Change owed: ");

        // Check quarters
        if (change >= 25)
        {
            quarters = change / 25;
            change = change - quarters * 25;
        }

        // Check dimes
        if (change < 25 && change >= 10)
        {
            dimes = change / 10;
            change = change - dimes * 10;
        }

        // Check nickels
        if (change < 10 && change >= 5)
        {
            nickels = change / 5;
            change = change - nickels * 5;
        }

        // Check pennies
        if (change < 5 && change >= 1)
        {
            pennies = change;
        }

        // Calculate the amount of coins
        int amount = quarters + dimes + nickels + pennies;
        printf("%i\n", amount);
    }
    while (change < 0);

    return 0;
}